Compiling breaks with rb-sys-build v0.9.65

I was trying to make a simple example gem and ran into this with the `bundle gem --ext=rust ...` scaffolding.

I was able to reproduce with a minimal `Cargo.toml`

This builds fine with 0.9.64.

```toml
[package]
name = "rb-sys-test"
version = "0.1.0"
edition = "2021"

[dependencies]
rb-sys-build = "=0.9.64"
```

```
╰─➤  rustup run 1.66 cargo build
    Updating crates.io index
  Downloaded rb-sys-build v0.9.64
  Downloaded 1 crate (14.8 KB) in 0.60s
   Compiling proc-macro2 v1.0.52
   Compiling quote v1.0.26
   Compiling bindgen v0.60.1
   Compiling rb-sys-build v0.9.64
   Compiling rb-sys-test v0.1.0 (/home/kmack/code/rb-sys-test)
    Finished dev [unoptimized + debuginfo] target(s) in 4.73s
```

When switching to 0.9.65 or higher

```toml
[package]
name = "rb-sys-test"
version = "0.1.0"
edition = "2021"

[dependencies]
rb-sys-build = "=0.9.65"
```

```
╰─➤  cargo build
    Updating crates.io index
   Compiling rb-sys-build v0.9.65
error[E0432]: unresolved import `quote::__private::Literal`
 --> /home/kmack/.cargo/registry/src/github.com-1ecc6299db9ec823/rb-sys-build-0.9.65/src/bindings/sanitizer.rs:1:24
  |
1 | use quote::__private::{Literal, TokenTree};
  |                        ^^^^^^^ no `Literal` in `__private`

error[E0603]: enum `TokenTree` is private
 --> /home/kmack/.cargo/registry/src/github.com-1ecc6299db9ec823/rb-sys-build-0.9.65/src/bindings/sanitizer.rs:1:33
  |
1 | use quote::__private::{Literal, TokenTree};
  |                                 ^^^^^^^^^ private enum
  |
note: the enum `TokenTree` is defined here
 --> /home/kmack/.cargo/registry/src/github.com-1ecc6299db9ec823/quote-1.0.26/src/runtime.rs:6:49
  |
6 | use proc_macro2::{Group, Ident, Punct, Spacing, TokenTree};
  |                                                 ^^^^^^^^^

Some errors have detailed explanations: E0432, E0603.
For more information about an error, try `rustc --explain E0432`.
error: could not compile `rb-sys-build` due to 2 previous errors
```

I tried to compile with the latest 3 minor Rust versions as well and got the same results.

```
╰─➤  rustup show
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/kmack/.rustup

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu (default)
1.66-x86_64-unknown-linux-gnu
1.67-x86_64-unknown-linux-gnu

active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.68.0 (2c8cc3432 2023-03-06)
```

Ah looks like it's an issue with `quote` dependency. Pinning `quote` to 1.0.24 fixes the build.

```toml
[package]
name = "rb-sys-test"
version = "0.1.0"
edition = "2021"

[dependencies]
quote = "=1.0.24"
rb-sys-build = "=0.9.65"
```

```
╰─➤  cargo build
    Updating crates.io index
   Compiling memchr v2.5.0
   Compiling glob v0.3.1
   Compiling proc-macro2 v1.0.52
   Compiling libc v0.2.140
   Compiling quote v1.0.24
   Compiling unicode-ident v1.0.8
   Compiling minimal-lexical v0.2.1
   Compiling cfg-if v1.0.0
   Compiling libloading v0.7.4
   Compiling syn v1.0.109
   Compiling regex-syntax v0.6.28
   Compiling bindgen v0.60.1
   Compiling lazycell v1.3.0
   Compiling lazy_static v1.4.0
   Compiling shlex v1.1.0
   Compiling peeking_take_while v0.1.2
   Compiling clang-sys v1.6.0
   Compiling nom v7.1.3
   Compiling aho-corasick v0.7.20
   Compiling bitflags v1.3.2
   Compiling rustc-hash v1.1.0
   Compiling shell-words v1.1.0
   Compiling regex v1.7.1
   Compiling cexpr v0.6.0
   Compiling rb-sys-build v0.9.65
   Compiling rb-sys-test v0.1.0 (/home/kmack/code/rb-sys-test)
    Finished dev [unoptimized + debuginfo] target(s) in 9.79s
```

Breaks as of quote 1.0.26 (1.0.25 appears to be missing)

```toml
[package]
name = "rb-sys-test"
version = "0.1.0"
edition = "2021"

[dependencies]
quote = "=1.0.26"
rb-sys-build = "=0.9.65"
```

```
╰─➤  cargo build
    Updating crates.io index
   Compiling quote v1.0.26
   Compiling bindgen v0.60.1
   Compiling syn v1.0.109
   Compiling rb-sys-build v0.9.65
error[E0432]: unresolved import `quote::__private::Literal`
 --> /home/kmack/.cargo/registry/src/github.com-1ecc6299db9ec823/rb-sys-build-0.9.65/src/bindings/sanitizer.rs:1:24
  |
1 | use quote::__private::{Literal, TokenTree};
  |                        ^^^^^^^ no `Literal` in `__private`

error[E0603]: enum `TokenTree` is private
 --> /home/kmack/.cargo/registry/src/github.com-1ecc6299db9ec823/rb-sys-build-0.9.65/src/bindings/sanitizer.rs:1:33
  |
1 | use quote::__private::{Literal, TokenTree};
  |                                 ^^^^^^^^^ private enum
  |
note: the enum `TokenTree` is defined here
 --> /home/kmack/.cargo/registry/src/github.com-1ecc6299db9ec823/quote-1.0.26/src/runtime.rs:6:49
  |
6 | use proc_macro2::{Group, Ident, Punct, Spacing, TokenTree};
  |                                                 ^^^^^^^^^

Some errors have detailed explanations: E0432, E0603.
For more information about an error, try `rustc --explain E0432`.
error: could not compile `rb-sys-build` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
```

--- a/crates/rb-sys-build/Cargo.toml
+++ b/crates/rb-sys-build/Cargo.toml
@@ -20,6 +20,7 @@ cc-impl = { version = "1.0", optional = true, package = "cc" }
 syn = { version = "1.0", features = ["parsing", "full", "extra-traits"] }
 quote = "1.0.23"
 lazy_static = "1.4.0"
+proc-macro2 = "1.0.52"
 
 [features]
 default = []

--- a/crates/rb-sys-build/src/bindings/sanitizer.rs
+++ b/crates/rb-sys-build/src/bindings/sanitizer.rs
@@ -1,7 +1,8 @@
-use quote::__private::{Literal, TokenTree};
+use proc_macro2::{Literal, TokenTree};
+use quote::ToTokens;
 use regex::Regex;
 use std::{borrow::Cow, error::Error};
-use syn::{Attribute, Item, __private::ToTokens};
+use syn::{Attribute, Item};
 
 lazy_static::lazy_static! {
     static ref URL_REGEX: Regex = Regex::new(r#"https?://[^\s'"]+"#).unwrap();

